bids/grabbids/bids_layout.py

```diff
@@ -63,10 +49,10 @@ def get_metadata(self, path):
     def get_fieldmap(self, path):
         sub = os.path.split(path)[1].split("_")[0].split("sub-")[1]
         fieldmap_set = {}
-        for file in self.get(subject=sub,
-                             type='(phase1|phase2|phasediff|epi|fieldmap)',
+        type_ = '(phase1|phase2|phasediff|epi|fieldmap)'
+        for file in self.get(subject=sub, type=type_,
                              extensions=['nii.gz', 'nii']):
-            metadata = self.get_metadata(file.filename)
+            metadata = self.get_metadata(file.filename, type=type_)
```

Comment:
Oh, maybe the thing that's not clear from this, and may be confusing, is that the first call (to `get()`) doesn't constrain the second one (to `get_metadata()`). Internally, `get_nearest` looks at *all* the files in the current project, not just the ones that were selected with `get`. It might make sense to take an argument in `get_nearest` that limits to certain files, but that would be functionally equivalent to what's going on here, because extra args in `get_nearest` (including `type`) are just passed through to the internal `get` call.